lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py

```diff
@@ -151,18 +147,34 @@
                     - Dynamic
                     - Static
                 default: Dynamic
+            load_balancer_backend_address_pools:
+                description:
+                    - List of an existing load-balancer backend address pool id to associate with the network interface.
+                    - It can be write as a resource id.
+                    - Also can be a dict of I(name) and I(load_balancer).
+                version_added: 2.6
             primary:
                 description:
                     - Whether the ip configuration is the primary one in the list.
                 type: bool
                 default: 'no'
         version_added: 2.5
-    security_group_name:
+    create_with_security_group:
```

Comment:
about this paramter, if no security_group specified, then a deault one should be created. it seems redundant to have a parameter to control the behavior

Reply:
Sure, but we cannot break current behavior: no nsg specified, create a default one

Reply:
does that mean there's scenario to create a nic without nsg? interesting

Reply:
yes, support user create a NIC without NSG. default behavior is create a default NSG and attach to NIC